akka-actor/src/main/scala/akka/serialization/Serializer.scala

```diff
@@ -135,6 +135,7 @@ abstract class SerializerWithStringManifest extends Serializer {
    * and message is dropped. Other exceptions will tear down the TCP connection
    * because it can be an indication of corrupt bytes from the underlying transport.
    */
+  @throws(classOf[NotSerializableException])
```

Comment:
good catch, that is in more places such as `JSerializer`, but I think you can add it to all `fromBinary` methods in this file